common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -288,82 +295,248 @@ func (v *pinotVisibilityTripleManager) chooseVisibilityManagerForWrite(ctx conte
 	}
 }
 
+// For Pinot Migration uses. It will be a temporary usage
+type userParameters struct {
+	operation    string
+	isOpen       bool
+	domainName   string
+	workflowType string
+	workflowID   string
+	closeStatus  int
+	customQuery  string
+}
+
+// For Pinot Migration uses. It will be a temporary usage
+// logUserQueryParameters will log user queries' parameters so that a comparator workflow can consume
+func (v *pinotVisibilityTripleManager) logUserQueryParameters(userParam userParameters) {
+	// Don't log if it is not enabled
+	if !v.logCustomerQueryParameter() {
+		return
+	}
+	randNum := rand.Intn(10)
+	if randNum != 5 { // Intentionally to have 1/10 chance to log custom query parameters
+		return
+	}
+
+	// filters start time range if user provided one
+	filteredQuery, err := v.queryFilter.FilterQuery(userParam.customQuery)
+	if err != nil {
+		v.logger.Error(fmt.Sprintf("Query filter error! %s", err))
+	}
+
+	v.logger.Info("Logging user query parameters for Pinot/ES response comparator...",
+		tag.OperationName(userParam.operation),
+		tag.IsWorkflowOpen(userParam.isOpen),
+		tag.WorkflowDomainName(userParam.domainName),
+		tag.WorkflowType(userParam.workflowType),
+		tag.WorkflowID(userParam.workflowID),
+		tag.WorkflowCloseStatus(userParam.closeStatus),
+		tag.VisibilityQuery(filteredQuery))
+}
+
 func (v *pinotVisibilityTripleManager) ListOpenWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:    "list",
+		isOpen:       true,
+		domainName:   request.Domain,
+		workflowType: "",
+		workflowID:   "",
+		closeStatus:  math.MinInt, // set the default close status to be Integer.MIN_VALUE
+		customQuery:  "",
+	})
+
 	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	override := ctx.Value("visibility-override")
```

Comment:
We can write in one statement 
```
if override := ctx.Value("visibility-override"); override == "ES" {
		manager = v.esVisibilityManager
}
```

Reply:
Currently the default reader is ES, the logic here is the reverse way.

Reply:
"ES" and "visibility-override" could put in constant string so in the future you can change them easily

Reply:
Maybe change like this, so it can work no matter what default option is, and even if it is not set override, it still works
```
if override := ctx.Value("visibility-override"); override == "ES" {
        manager = v.esVisibilityManager
} else if override == "Pinot" {
        manager = v.pinotVisibilityManager
}
```

Reply:
do we really need to set override value for both cases? I was thinking to only have the attribute `override` when we really need to "override" something.

Reply:
Before we fully switch to Pinot, it will read from ES by fault, so we need if override == "Pinot" case. I think the question will if we need to keep this comparator running after we switch to Pinot. If so, we gonna need  if override == "es" case.